### PR TITLE
feat(remix-dev): use native pnp resolution in esbuild 0.15+

### PR DESCRIPTION
**Pull request by redabacha**
Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

**Comment by mcansh** [comment]
i was hoping for this PR 🤩 - i know nothing about pnp though, so I'll defer to @pcattori - but if it's good, perhaps we can bump up to 0.15.9 to also include the jsx name collision fixes :)

File: packages/remix-dev/package.json [hunk 1]
```diff
@@ -25,13 +25,12 @@
...
     "@npmcli/package-json": "^2.0.0",
     "@remix-run/server-runtime": "1.7.0",
-    "@yarnpkg/esbuild-plugin-pnp": "3.0.0-rc.11",
     "arg": "^5.0.1",
     "cacache": "^15.0.5",
     "chalk": "^4.1.2",
     "chokidar": "^3.5.1",
     "dotenv": "^16.0.0",
-    "esbuild": "0.14.51",
+    "esbuild": "0.15.7",
```
Comment (MichaelDeBoey) [inline packages/remix-dev/package.json:33]:
```suggestion
    "esbuild": "0.15.10",
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Comment by MichaelDeBoey** [comment]
@redabacha Could you please rebase your branch onto latest `dev` & resolve conflicts?

File: packages/remix-dev/package.json (hunk 1 above)
Comment (redabacha) [reply packages/remix-dev/package.json]: updated :sweat_smile:

**Review by pcattori — APPROVED** [review]

**Comment by pcattori** [comment]
@mcansh : let's do some solid testing on the `esbuild` upgrade before we merge this

**Comment by mcansh** [comment]
100%

**Review by MichaelDeBoey — APPROVED** [review]

**Comment by mattste** [comment]
I just upgraded my team's very small internal app to this P/R and no issues thus far. I'll update if we run into anything.

**Comment by machour** [comment]
This have been implemented in #4301 
Thank you @redabacha!

**Pull request closed without merging**